diff --git a/CMakeLists.txt b/CMakeLists.txt
index ac3e471..dec4ee6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,6 +8,8 @@ include_directories(${RIME_SOURCE_DIR}/deps/leveldb/include)
 find_package(SQLite3 REQUIRED)

 add_library(rime-predict-objs OBJECT ${predict_src})
+target_compile_options(rime-predict-objs PRIVATE "-ffile-prefix-map=${CMAKE_SOURCE_DIR}=.")
+target_compile_definitions(rime-predict-objs PRIVATE BOOST_ALL_NO_EMBEDDED_GDB_SCRIPTS)
 if(BUILD_SHARED_LIBS)
   set_target_properties(rime-predict-objs
     PROPERTIES
