diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4c2fcff6..e3109437 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -5,10 +5,20 @@ aux_source_directory(src predict_src)
 include_directories(deps/msgpack-c/include)
 include_directories(${RIME_SOURCE_DIR}/deps/leveldb/include)
 
-find_package(SQLite3 REQUIRED)
-include_directories(${SQLite3_INCLUDE_DIRS})
+if(ANDROID)
+  add_library(rime-predict-sqlite3 STATIC
+    ${RIME_SOURCE_DIR}/deps/sqlite3/sqlite3.c)
+  target_include_directories(rime-predict-sqlite3 PUBLIC
+    ${RIME_SOURCE_DIR}/deps/sqlite3)
+  include_directories(${RIME_SOURCE_DIR}/deps/sqlite3)
+else()
+  find_package(SQLite3 REQUIRED)
+  include_directories(${SQLite3_INCLUDE_DIRS})
+endif()
 
 add_library(rime-predict-objs OBJECT ${predict_src})
+target_compile_options(rime-predict-objs PRIVATE "-ffile-prefix-map=${CMAKE_SOURCE_DIR}=.")
+target_compile_definitions(rime-predict-objs PRIVATE BOOST_ALL_NO_EMBEDDED_GDB_SCRIPTS)
 if(BUILD_SHARED_LIBS)
   set_target_properties(rime-predict-objs
     PROPERTIES
@@ -17,5 +27,10 @@ endif()
 
 set(plugin_name rime-predict PARENT_SCOPE)
 set(plugin_objs $<TARGET_OBJECTS:rime-predict-objs> PARENT_SCOPE)
-set(plugin_deps ${rime_library} leveldb sqlite3 PARENT_SCOPE)
+if(ANDROID)
+  set(predict_sqlite_lib rime-predict-sqlite3)
+else()
+  set(predict_sqlite_lib sqlite3)
+endif()
+set(plugin_deps ${rime_library} leveldb ${predict_sqlite_lib} PARENT_SCOPE)
 set(plugin_modules "predict" PARENT_SCOPE)
